src/data/list/basic.lean

```diff
@@ -4552,6 +4552,37 @@ theorem prod_range_succ {α : Type u} [monoid α] (f : ℕ → α) (n : ℕ) :
 by rw [range_concat, map_append, map_singleton,
   prod_append, prod_cons, prod_nil, mul_one]
 
+/-- A variant of `prod_range_succ` which pulls off the first
+    term in the product rather than the last.-/
+@[to_additive "A variant of `sum_range_succ` which pulls off the first term in the sum
+  rather than the last."]
+theorem prod_range_succ' {α : Type u} [monoid α] (f : ℕ → α) (n : ℕ) :
+  ((range n.succ).map f).prod = f 0 * ((range n).map (λ i, f (succ i))).prod :=
+begin
+  induction n with d hd,
+  { show 1 * f 0 = f 0 * 1, rw [one_mul, mul_one]},
+  { rw [list.prod_range_succ, hd, mul_assoc, ←list.prod_range_succ]},
+end
+
+@[to_additive]
+theorem prod_map_hom {α β γ : Type*} [monoid β] [monoid γ] (L : list α) (f : α → β) (g : β →* γ) :
+  (L.map (g ∘ f)).prod = g ((L.map f).prod) :=
```

Comment:
I'd use `map_map` + `monoid_hom.map_list_prod`.

Reply:
I've done this, but `map_list_prod` isn't defined until later on so I've moved things, which might confuse github.